src/components/calendar/MedicalCalendar.tsx

```diff
@@ -1,43 +1,51 @@
 import React, {useContext, useEffect} from "react";
-import Calendar from "react-calendar";
+
 import moment from "moment";
-import {MedicalCalendarContext} from "../../context/calendar-context/MedicalCalendarContext";
+import {MedicalCalendarContext} from "src/context/calendar-context/MedicalCalendarContext";
 import 'react-calendar/dist/Calendar.css';
 import EventBlock from "../day-events/EventBlock";
 import './Calendar.css'
+import {CalendarContextType, CalendarEvent} from "src/models/Models";
+import Calendar from "react-calendar";
+
+type Props = {
+  selectedDate: Date,
+  selectDate: Function
+}
 
-const MedicalCalendar = ({selectedDate, selectDate}) => {
-  const [mcContext, setMcContext] = useContext(MedicalCalendarContext)
+const MedicalCalendar = ({selectedDate, selectDate} :Props) => {
+  const {events, saveEvents} = useContext(MedicalCalendarContext) as CalendarContextType
```

Comment:
Uma dica é criar um custom hook que já te devolve o contexto, assim não precisa chamar o useContext passando o seu contexto como parâmetro sempre.

https://kentcdodds.com/blog/how-to-use-react-context-effectively#the-custom-consumer-hook